src/meta/mod.rs

```diff
@@ -48,7 +48,8 @@ impl Meta {
         };
 
         let file_type = FileType::from(&metadata);
-        let name = Name::new(&path, file_type);
+        let permissions = Permissions::from(&metadata);
```

Comment:
If you initialize permissions here, you should use the variable for the Meta initialization below.

l.58 -- `permissions: Permissions::from(&metadata),`
l.58 ++ `permissions,`